scripts/anchor-test.sh

```diff
@@ -1,19 +1,24 @@
-#!/bin/sh
+#!/usr/bin/env bash
+set -euo pipefail
 
-echo "🔨 Building anchor programs (with test feature, no IDL)..."
-anchor build --no-idl -- --features test
+echo "🔨 Deterministic build (test feature, no IDL)…"
+anchor build --verifiable --no-idl -- --features test
 
 echo "📦 Generating IDLs (using nightly)..."
 
 for program in programs/*; do
   [ -d "$program" ] || continue
 
   dir_name=$(basename "$program")
-  program_name=$(echo "$dir_name" | tr '-' '_')
+  program_name=${dir_name//-/_}
 
   echo "→ IDL for $program_name"
-  RUSTUP_TOOLCHAIN="nightly-2025-04-01" anchor idl build -p "$dir_name" -o "target/idl/$program_name.json" -t "target/types/$program_name.ts"
+  RUSTUP_TOOLCHAIN="nightly-2025-04-01" \
+    anchor idl build \
+      -p "$dir_name" \
+      -o "target/idl/$program_name.json" \
+      -t "target/types/$program_name.ts"
 done
 
-echo "🧪 Running anchor tests..."
-anchor test --skip-build
+echo "🧪 Running deterministic tests (test feature, no IDL)…"
+anchor test --verifiable --no-idl -- --features test
```

Comment:
does test has these flags? I think here we should just leave `--skip-build` as built should have been performed above